toolbox/ui.py

```diff
@@ -136,7 +136,28 @@ def draw_umap_projections(self, utterances: Set[Utterance]):
         self.umap_ax.set_xticks([])
         self.umap_ax.set_yticks([])
         self.umap_ax.figure.canvas.draw()
-        
+
+    def setup_audio_devices(self,sample_rate):
+        supported_devices=[]
+        for device in sd.query_devices():
+            try:
+                sd.check_output_settings(device=device['name'], samplerate=sample_rate)
+                supported_devices.append(device['name'])
+            except Exception as e:
+                warn("Unsupported device "+ device['name']+" for the sample rate: "+str(sample_rate)+": "+str(e))
+        if len(supported_devices)==0:
+            error_dialog = QtWidgets.QErrorMessage()
```

Comment:
This line throws an error. It should be `error_dialog = QErrorMessage()` because line 4 has `from PyQt5.QtWidgets import *`. Also on my system there is no pop-up or message display when this code is executed. Please try to test the error cases when adding them.

In the pull request to your fork, I switched to using self.log() to display the error message.